main.go

```diff
@@ -77,12 +77,12 @@ func listInternalSources(limit, offset int64) *SourceResponse {
 	log.Printf("Requesting [limit %v] + [offset %v] sources from internal API at [%v]", limit, offset, host)
 
 	url, _ := url.Parse(fmt.Sprintf("%v/internal/v2.0/sources?limit=%v&offset=%v", host, limit, offset))
-	req := &http.Request{Method: "GET", URL: url, Header: map[string][]string{
+	req := &http.Request{Method: http.MethodGet, URL: url, Header: map[string][]string{
 		"x-rh-sources-account-number": {"sources_monitor"},
 		"x-rh-sources-psk":            {psk},
 	}}
 	resp, err := httpClient.Do(req)
-	if err != nil || resp.StatusCode != 200 {
+	if err != nil || (resp != nil && resp.StatusCode != 200) {
```

Comment:
I've checked the docs for the `.Do` function, and it states the following:

```go
// An error is returned if caused by client policy (such as
// CheckRedirect), or failure to speak HTTP (such as a network
// connectivity problem). A non-2xx status code doesn't cause an
// error.

// If the returned error is nil, the Response will contain a non-nil
// Body which the user is expected to close. If the Body is not both
// read to EOF and closed, the Client's underlying RoundTripper
// (typically Transport) may not be able to re-use a persistent TCP
// connection to the server for a subsequent "keep-alive" request.
```
and
```go
// On error, any Response can be ignored. A non-nil Response with a
// non-nil error only occurs when CheckRedirect fails, and even then
// the returned Response.Body is already closed.
````

So I'm not sure if we will ever get a `err != nil` with a `nil` response. Maybe a plain TCP endpoint which doesn't talk HTTP?

Reply:
Yeah the error is very _VERY_ intermittent too - it just appears to happen sometimes when the deployment is being rolled out I think. I just know that it definitely `panic`'s due to a nil pointer somewhere - so trying to prevent that. 

EDIT: It should be as the pod is spinning up the server isn't serving HTTP yet since the pod is there but puma isn't started - but it squawks TCP. But the readiness probe should prevent that. So yeah its all kind of weird.

Reply:
All right, understood!